### ISSUE TITLE
Problem with runModEclipse/IDEA and auto-refresh

### ISSUE THREAD

**Issue by ddossot** [crashub/mod-shell#4]
When vert.x refreshes a loaded mod in development mode, Crash crashes because the ports it needs are already taken by the previous instance of itself.

I'm wondering if this is because only `start()` methods are implemented in the code that ties Crash to Vert.x? Should `stop()` methods be added in order to perform a clean shutdown of Crash when Vert.x recycles the mod that deploys it?

**Comment by vietj** [comment]
can you tell me how to reproduce ? I never tried vert.x in dev mode so far

**Comment by ddossot** [comment]
- Checkout the quick-start: http://vertx.io/gradle_dev.html
- Add the following in GroovyPingVerticle:
  
  ```
  container.deployModule('org.crashub~vertx.shell~2.0.2', [
      'cmd': '../commands',
      'crash.auth': 'simple',
      'crash.auth.simple.username': 'admin',
      'crash.auth.simple.password': 'admin',
      'crash.ssh.port': 2000
  ])
  ```
- Edit mod.json and ensure that the Groovy verticle is the main and that the project is auto-redeploying:
  <pre>
    "main":"groovy:com.mycompany.myproject.GroovyPingVerticle",
    "auto-redeploy": true,
  </pre>
- Start vert.x from the command line with `./gradlew runModIDEA` (or runModEclipse for small players like who use Eclipse)
- Connect to Crash and ensure it works
- Disconnect
- Make a small change to GroovyPingVerticle and save.
- Vert.x auto-redeploys the application but:
  <pre>
  Could not start CRaSSHD
  java.net.BindException: Address already in use
  at sun.nio.ch.Net.bind0(Native Method)
  at sun.nio.ch.Net.bind(Net.java:344)
  at sun.nio.ch.Net.bind(Net.java:336)
  at sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:199)
  at sun.nio.ch.ServerSocketAdaptor.bind(ServerSocketAdaptor.java:74)
  at org.apache.mina.transport.socket.nio.NioSocketAcceptor.open(NioSocketAcceptor.java:201)
  at org.apache.mina.transport.socket.nio.NioSocketAcceptor.open(NioSocketAcceptor.java:51)
  at org.apache.mina.core.polling.AbstractPollingIoAcceptor.registerHandles(AbstractPollingIoAcceptor.java:543)
  at org.apache.mina.core.polling.AbstractPollingIoAcceptor.access$300(AbstractPollingIoAcceptor.java:67)
  at org.apache.mina.core.polling.AbstractPollingIoAcceptor$Acceptor.run(AbstractPollingIoAcceptor.java:418)
  at org.apache.mina.util.NamePreservingRunnable.run(NamePreservingRunnable.java:64)
  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1110)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:603)
  at java.lang.Thread.run(Thread.java:722)
  </pre>
- Adding the following to the Groovy verticle doesn't help:
  <pre>
  def stop() {
      container.undeployModule('org.crashub~vertx.shell~2.0.2')
  }
  </pre>

**Comment by vietj** [comment]
it is fixed and a 2.0.3 has been released, it should hit maven central today.
